src/services/api.requests.ts

```diff
@@ -1,59 +1,55 @@
-import { getClientLocalization } from "./client-localization.request";
-import { RequestProps, FetchResult, ClimateDetails } from "./services.inteface";
+import { RequestProps, FetchResult, ClimateDetails } from './services.inteface';
 
-export const corsAnywhere = "https://cors-anywhere.herokuapp.com/";
+export const corsAnywhere = 'https://cors-anywhere.herokuapp.com/';
 
-//TODO: Make an unique function?
+function makeRequest(link: string): Promise<any> {
+  return new Promise((resolve, reject) => {
+    fetch(corsAnywhere + link)
+      .then((result: Response) => {
+        result.ok ? resolve(result.json()) : alert(result.statusText);
+      })
+      .catch(reject);
+  });
+}
 
-export async function RequestByName(
-  props: RequestProps
-): Promise<FetchResult[] | void> {
+export function requestByName(props: RequestProps): Promise<FetchResult[] | void> {
   const APILink = `http://www.metaweather.com/api/location/search/?query=${props.location}`;
-  try {
-    const result = await fetch(corsAnywhere + APILink);
-    console.log(result);
-    try {
-      const data: FetchResult[] = await result.json();
-      console.log(data);
-      return data;
-    } catch (error) {
-      console.log("JSON Error:", error);
-    }
-  } catch (error) {
-    console.log("Error:", error);
-  }
+  return new Promise((resolve, reject) => {
+    makeRequest(APILink).then(resolve).catch(reject);
+  });
```

Comment:
There is no need to return a new Promise here. Same to other occurrences below.

```suggestion
  return makeRequest(APILink);
```